### PR TITLE
Replacing ManifestFile Path property with []byte content

### PR DESCRIPTION
**Pull request by manasachi**
# Description

Replacing ManifestFile Path property with yaml content as []byte. There is no need to write rendered manifests to a temporary directory now. Manifests are rendered and written in memory for validation to handle later.

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Refactor

# How Has This Been Tested?
Unit tests were updated to ensure validation is not dependent on writing rendered files to a temp dir. All files are written/returned in memory

- [ ] Test A
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/safeguards/helpers.go [hunk 1]
```diff
@@ -127,9 +127,55 @@ func GetManifestFiles(p string) ([]c.ManifestFile, error) {
...
+			log.Debugf("%s is a directory, skipping...", info.Name())
+		}
 
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not walk directory: %s", err)
+	}
+	if len(manifestFiles) == 0 {
+		return nil, fmt.Errorf(" found within given path")
```
Comment (NickKeller) [inline pkg/safeguards/helpers.go:149]: should probably be "no manifest files found within given path"?

File: pkg/safeguards/helpers.go [hunk 2]
```diff
@@ -111,11 +111,11 @@ func IsYAML(path string) bool {
 }
 
 // GetManifestFiles uses filepath.Walk to retrieve a list of the manifest files within the given manifest path
-func GetManifestFiles(p string) ([]c.ManifestFile, error) {
-	var manifestFiles []c.ManifestFile
+func GetManifestFiles(p string) ([]types.ManifestFile, error) {
```
Comment (NickKeller) [inline pkg/safeguards/helpers.go:114]: This function is unused, you can remove it

**Review by NickKeller — CHANGES_REQUESTED** [review]

File: pkg/safeguards/helpers.go (hunk 1 above)
Comment (manasachi) [reply pkg/safeguards/helpers.go]: Good catch, thank you!

File: pkg/safeguards/helpers.go (hunk 2 above)
Comment (manasachi) [reply pkg/safeguards/helpers.go]: Seemed to have duplicated it, removed now.

**Review by NickKeller — APPROVED** [review]

File: pkg/safeguards/preprocessing/preprocessing_test_helpers.go [hunk 3]
```diff
@@ -31,34 +30,34 @@ const (
...
+	return yamlFileContent
 }
 
-func getManifestAsString(t *testing.T, filePath string) string {
-	yamlFileContent, err := os.ReadFile(filePath)
-	if err != nil {
-		t.Fatalf("Failed to read YAML file: %s", err)
-	}
+// replace newlines with strings for easy .yaml byte comparison
+func normalizeNewlines(data []byte) []byte {
```
[the remark this answers is not in the document]
Comment (manasachi) [reply pkg/safeguards/preprocessing/preprocessing_test_helpers.go:44]: Updated the function a little to make it as simple as possible without failing test cases

**Pull request closed without merging**